system/Database/BaseBuilder.php

```diff
@@ -2343,13 +2360,11 @@ protected function _truncate($table)
 	 *
 	 * @return string
 	 */
-	public function getCompiledDelete($reset = true)
+	public function getCompiledDelete(bool $reset = true): string
 	{
 		$table = $this->QBFrom[0];
 
-		$this->returnDeleteSQL = true;
-		$sql                   = $this->delete($table, '', null, $reset);
-		$this->returnDeleteSQL = false;
+		$sql = $this->delete($table, '', null, $reset);
```

Comment:
While I agree with removing the wrapping lines as not neccessary - the default value for `$returnSQL` in the `delete()` method is false, and this function is expecting it to be returned. 

You need to pass `true` as the 4th argument here.